### PR TITLE
Feat: new upload files

### PR DESCRIPTION
**Pull request by maxwelbm**
Test upload s3 zip

### CODE REVIEW

**Review by rodrigomaria — DISMISSED** [review]

File: pkg/cmd/deploy/upload.go [hunk 1]
```diff
@@ -1,136 +1,282 @@
...
+		SigningRegion: region,
+	}
 
+	cfg, err := config.LoadDefaultConfig(context.TODO(),
+		config.WithRegion(region),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(settings.S3AccessKey, settings.S3SecreKey, "")),
+		config.WithEndpointResolver(endpointResolver), // nolint
+	)
+	if err != nil {
+		return errors.New("unable to load SDK config, " + err.Error())
```
Comment (PatrickMenoti) [inline pkg/cmd/deploy/upload.go:240]: Put this message in our errors.go file

File: pkg/cmd/deploy/upload.go [hunk 2]
```diff
@@ -1,136 +1,282 @@
...
-		if info.IsDir() && (strings.Contains(path, "node_modules") || strings.Contains(path, ".edge")) {
-			logger.Debug("Skipping directory: " + path)
-			return filepath.SkipDir
+	for file := range filesChan {
+		filesToZip = append(filesToZip, file)
+
+		// Get the file size and add it to the current size
+		fileInfo, err := os.Stat(file)
+		if err != nil {
+			log.Printf("Failed to get file info for %s: %v", file, err)
```
Comment (PatrickMenoti) [inline pkg/cmd/deploy/upload.go:144]: Change to logger.Debug or logger.Error

File: pkg/cmd/deploy/upload.go [hunk 3]
```diff
@@ -1,136 +1,282 @@
...
-		if !info.IsDir() {
-			fileContent, err := cmd.Open(path)
+		// If the current size exceeds the limit (1MB), zip and upload
+		if currentSize >= maxZipSize {
+			err = zipAndUpload(ctx, cfg, filesToZip, zipFileName, baseDir,
+				atomicCounter, bucketName, progressBar)
 			if err != nil {
-				logger.Debug("Error while trying to read file <"+path+"> about to be uploaded", zap.Error(err))
-				return err
+				log.Printf("Failed to zip and upload files: %v", err)
```
Comment (PatrickMenoti) [inline pkg/cmd/deploy/upload.go:154]: Change this to a variable and return the error variable
Also: you SHOULD return the error, to stop the execution :)

File: pkg/cmd/deploy/upload.go [hunk 4]
```diff
@@ -1,136 +1,282 @@
...
+			filesToZip = nil
+		}
+	}
 
-			Jobs <- fileOptions
+	// If there are still files at the end, zip them up and upload them
+	if len(filesToZip) > 0 {
+		err := zipAndUpload(ctx, cfg, filesToZip, zipFileName, baseDir, atomicCounter, bucketName, progressBar)
+		if err != nil {
+			log.Printf("Failed to zip and upload files: %v", err)
```
Comment (PatrickMenoti) [inline pkg/cmd/deploy/upload.go:167]: Change this to a variable and return the error variable
Also: you SHOULD return the error, to stop the execution :)

File: pkg/cmd/deploy/upload.go [hunk 5]
```diff
@@ -1,136 +1,282 @@
...
+	}
 
-		if bar != nil {
-			err := bar.Set(int(currentFile))
-			if err != nil {
-				return err
-			}
+	err = zipWriter.Close()
+	if err != nil {
+		return fmt.Errorf("failed to close zip file %s: %w", zipFileName, err)
```
Comment (PatrickMenoti) [inline pkg/cmd/deploy/upload.go:200]: Add this to our errors.go messages file and return the variable

File: pkg/cmd/deploy/upload.go [hunk 6]
```diff
@@ -1,136 +1,282 @@
...
+	}
+
+	// Check if the zip file was created
+	if _, err := os.Stat(zipFilePath); os.IsNotExist(err) {
+		return fmt.Errorf("zip file %s does not exist", zipFileName)
```
Comment (PatrickMenoti) [inline pkg/cmd/deploy/upload.go:205]: Add this to our errors.go messages file and return the variable

File: pkg/cmd/deploy/upload.go [hunk 7]
```diff
@@ -1,136 +1,282 @@
...
+	}
+
+	err = uploadFile(ctx, cfg, bucketName, zipFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to upload zip file %s: %w", zipFileName, err)
```
Comment (PatrickMenoti) [inline pkg/cmd/deploy/upload.go:210]: Add this to our errors.go messages file and return the variable

**Review by PabloDiehl — APPROVED** [review]

**Review by rodrigomaria — APPROVED** [review]

**Pull request merged**